programs/drift/src/instructions/keeper.rs

```diff
@@ -475,6 +478,93 @@ pub fn handle_settle_pnl(ctx: Context<SettlePNL>, market_index: u16) -> Result<(
     Ok(())
 }
 
+#[access_control(
+settle_pnl_not_paused(&ctx.accounts.state)
+)]
+pub fn handle_settle_pnls(
+    ctx: Context<SettlePNL>,
+    market_indexes: Vec<u16>,
+    mode: SettlePnlMode,
+) -> Result<()> {
+    let clock = Clock::get()?;
+    let state = &ctx.accounts.state;
+
+    let user_key = ctx.accounts.user.key();
+    let user = &mut load_mut!(ctx.accounts.user)?;
+
+    let AccountMaps {
+        perp_market_map,
+        spot_market_map,
+        mut oracle_map,
+    } = load_maps(
+        &mut ctx.remaining_accounts.iter().peekable(),
+        &get_writable_perp_market_set_from_vec(&market_indexes),
+        &get_writable_spot_market_set(QUOTE_SPOT_MARKET_INDEX),
+        clock.slot,
+        Some(state.oracle_guard_rails),
+    )?;
+
+    let meets_margin_requirement = meets_maintenance_margin_requirement(
```

Comment:
do we need to call update amm on every market before calling this?

Reply:
many other maint checks dont require update_amm